### PR TITLE
Update astyle to 2.06

### PR DESCRIPTION
**Pull request by 3nids**
Following discussion in #3463 this updates astyle to 2.06.
I gave it a run on src/core, and changes seems quite reasonable.
There is a bit of noise with all the multi-line initializer lists shifted to the left by 2.

If we agree on updating, I left some options commented and some shall be discussed (maybe afterwards)

* [x] position of pointer
* [x] position of reference
* [ ] remove brackets for one-line ifs
* [x] padding of parenthesis

As Matthias pointed out this makes the beautifying compatible with c++11 syntax.

### CODE REVIEW

**Comment by m-kuhn** [comment]
I think removing brackets for one-line ifs is fine for everyone

**Comment by 3nids** [comment]
@m-kuhn fine with me, I'll add this in a second step to allow better inspection of the diff. But indeed that means quite some lines removed :)

**Comment by 3nids** [comment]
I put back the parenthesis options. This leads to these modifs:

image
image

**Comment by 3nids** [comment]
To finalize this:

following different opinions expressed in #3463 and what seems to be recognized as a standard, let's use:
```
--align-pointer=name 
--align-reference=name
```

I am also removing brackets in one-line ifs.

**Comment by 3nids** [comment]
this is ready for approval/disapproval

File: src/core/effects/qgsimageoperation.cpp [hunk 1]
```diff
@@ -691,9 +657,7 @@ void QgsImageOperation::GaussianBlurOperation::operator()( QgsImageOperation::Im
       sourceRef = sourceFirstLine;
       destRef = reinterpret_cast< QRgb * >( outputLineRef );
       for ( int x = 0; x < width; ++x, ++destRef, sourceRef += 4 )
-      {
-        *destRef = gaussianBlurVertical( y, sourceRef, sourceBpl, height );
-      }
+        * destRef = gaussianBlurVertical( y, sourceRef, sourceBpl, height );
```
Comment (3nids) [inline src/core/effects/qgsimageoperation.cpp]: is this expected here?
would that cause any trouble?

Comment (m-kuhn) [inline src/core/effects/qgsimageoperation.cpp]: Hmmm... the asterisk is used to dereference in this situation, according to my preference it should be variable-aligned here.

Not sure if astyle is able to properly disambiguate.

Comment (3nids) [inline src/core/effects/qgsimageoperation.cpp]: It's either the combination of `remove_brackets` with either `pad-oper` (operator padding) or `align-pointer=name`. I would suspect it's with operator but will have to check and maybe report an error in astyle.

File: src/core/auth/qgsauthcertutils.cpp [hunk 2]
```diff
@@ -119,9 +117,7 @@ QList<QSslCertificate> QgsAuthCertUtils::certsFromFile( const QString &certspath
   bool pem = certspath.endsWith( QLatin1String( ".pem" ), Qt::CaseInsensitive );
   certs = QSslCertificate::fromData( fileData_( certspath, pem ), pem ? QSsl::Pem : QSsl::Der );
   if ( certs.isEmpty() )
-  {
     QgsDebugMsg( QString( "Parsed cert(s) EMPTY for path: %1" ).arg( certspath ) );
```
Comment (jef-n) [inline src/core/auth/qgsauthcertutils.cpp]: this is probably trouble, when building in release mode.

**Comment by nyalldawson** [comment]
@3nids Travis has issues at the moment - I can't get builds to run

**Comment by 3nids** [comment]
so it seems `remove_brackets` is bringing troubles at the moment, removing it.

**Comment by 3nids** [comment]
if no one objects, I'll merge this as is (i.e. without bracket removal)

**Pull request merged**